core/payment-driver/zksync/src/zksync/faucet.rs

```diff
@@ -94,3 +95,12 @@ async fn faucet_donate(address: &str, _network: Network) -> Result<(), GenericEr
     // TODO: Verify tx hash
     Ok(())
 }
+
+async fn resolve_faucet_url() -> Result<String, GenericError> {
+    match env::var(FAUCET_ADDR_ENVAR) {
+        Ok(addr) => Ok(addr),
```

Comment:
Also this `match` is here because compiler complains when awaiting in the closure (lambda) passed to `or_else`.
What would be better way to write this?

Reply:
You can try converting `env::var` result into [OptionFuture](https://docs.rs/futures/0.3.12/futures/future/struct.OptionFuture.html) and using asynchronous map function but I'm not sure if it's worth it.